plugins/modules/idrac_system_erase.py

```diff
@@ -0,0 +1,569 @@
+
+        :param idrac: The idrac parameter.
+        :type idrac: Any
+        :param module: The module parameter.
+        :type module: Any
+        """
+        self.idrac = idrac
+        self.module = module
+
+    def execute(self):
+        """
+        Executes the function with the given module.
+
+        :param module: The module to execute.
+        :type module: Any
+        :return: None
+        """
+
+    def get_system_erase_url(self):
+        """
+        Retrieves the URL for the system erase operation.
+
+        Returns:
+            str: The URL for the system erase operation.
+        """
+        url = self.get_url()
+        resp = get_dynamic_uri(self.idrac, url)
+        system_erase_url = (
+            resp.get('Links', {})
+            .get(OEM, {})
+            .get(MANUFACTURER, {})
+            .get(LC_SERVICE, {})
+            .get(ODATA, {})
+        )
+        return system_erase_url
+
+    def get_url(self):
+        """
+        Retrieves the URL for the resource.
+
+        Returns:
+            str: The URL for the resource.
+
+        Raises:
+            AnsibleExitJson: If the resource ID is not found.
+        """
+        res_id = self.module.params.get('resource_id')
+        if res_id:
+            uri = MANAGERS_URI + "/" + res_id
+        else:
+            uri, error_msg = validate_and_get_first_resource_id_uri(
+                self.module, self.idrac, MANAGERS_URI)
+            if error_msg:
+                self.module.exit_json(msg=error_msg, failed=True)
+        return uri
+
+    def get_job_status(self, erase_component_response):
+        """
+        Retrieves the status of a job.
+
+        Args:
+            erase_component_response (object): The response object for the erase component
+            operation.
+
+        Returns:
+            dict: The job details.
+
+        Raises:
+            AnsibleExitJson: If the job tracking times out.
+        """
+        job_wait_timeout = self.module.params.get('job_wait_timeout')
+        res_uri = validate_and_get_first_resource_id_uri(self.module, self.idrac, MANAGERS_URI)
+        job_tracking_uri = erase_component_response.headers.get("Location")
+        job_id = job_tracking_uri.split("/")[-1]
+        job_uri = IDRAC_JOB_URI.format(job_id=job_id, res_uri=res_uri[0])
+        job_failed, msg, job_dict, wait_time = idrac_redfish_job_tracking(
+            self.idrac, job_uri, max_job_wait_sec=job_wait_timeout)
+        job_dict = remove_key(job_dict, regex_pattern=ODATA_REGEX)
+        if int(wait_time) >= int(job_wait_timeout):
+            self.module.exit_json(msg=WAIT_TIMEOUT_MSG.format(
+                job_wait_timeout), changed=True, job_status=job_dict)
+        if job_failed:
+            self.module.exit_json(
+                msg=job_dict.get('Message'),
+                failed=True,
+                job_details=job_dict)
+        return job_dict
+
+    def get_job_details(self, erase_component_response):
```

Comment:
Can you please check if we have same function in utils if not please move it.

Reply:
Its not available in the utils. Keeping this in the module itself. Will take care of adding to utils in the upcoming release